### DISCUSSION TITLE
Is importing OVA/OVF Files possible?

### DISCUSSION THREAD

**Discussion by notmnky** [utmapp/UTM#2521] [UTM Help (macOS)]
Hi!
Up until now I was using VM Virtualbox, but after switching to an M1 i realized I can't work with that anymore. I am an absolute beginner, and need to run HADOOP/HIVE that has been provided to me in an OVA compilation.
Or is there any other way to export the file from Virtualbox to be able to open it in UTM?

**Comment by conath** [comment]
Hi notmnky, the file format used by UTM for VM hard disks is QCOW2 (UTM is based on QEMU). Other formats are technically supported but result in degraded performance and occasional data loss.

I found this guide online about converting from OVA to QCOW2. https://www.xmodulo.com/convert-ova-to-qcow2-linux.html

After you have converted the file, create a new VM in UTM with a configuration similar to the one you were using in VirtualBox. Save it and re-open the configuration (right click, edit or the button at the top right). In the Drives tab, click on Import Drive and select the QCOW2 file. The file will be copied to the VM bundle and appears in the list of drives. Now save the config and start the VM.

**Reply by Borjand** [reply]
Hi Conath, 

I am trying this with one of my VirtualBox machines (which is based on Debian and x86_64 precessing architecture), but when I try to start the VM, UTM crashes and I am not able to see what is causing this issue. Just to take into account, the VM starts, and once I select the Debian option in the initial GRUB screen, the UTM pops-up the following error: qemu-x86_64-softmmu: warning: Spice: playback:0 (0x7f9aad808a00): setsockopt failed, Operation not supported on socket.

Any suggestions on this?

Thanks in advance!

Borja

PS: I attach a pair of files that showcase what I mentioned above
<img width="1280" alt="screen-capture-1" src="https://user-images.githubusercontent.com/28540138/133250248-d06a7cfb-18ca-4b76-ae52-3ab5769d893f.png">

<img width="1280" alt="screen-capture-2" src="https://user-images.githubusercontent.com/28540138/133250278-842f3868-5953-4ac2-a065-a0a75cb89252.png">

**Reply by conath** [reply]
To troubleshoot this we need a debug log. To get it, go to the VM configuration => QEMU and enable debug logging checkbox. Then save & start the VM again. After the crash go back to the same QEMU page and click export debug log => mail => drag the file into GitHub.

**Reply by Borjand** [reply]
Thanks for the quick response @conath!

Please find below the debugging log:

Running:  -L /Applications/UTM.app/Contents/Resources/qemu -S -qmp tcp:127.0.0.1:4000,server,nowait -nodefaults -vga none -spice "unix=on,addr=/Users/borja/Library/Group Containers/WDNLXAD4W8.com.utmapp.UTM/81755D42-6831-410E-9FA8-A89B64FC1756.spice,disable-ticketing=on,image-compression=off,playback-compression=off,streaming-video=off,gl=on" -device virtio-vga-gl -cpu host -smp cpus=4,sockets=1,cores=2,threads=2 -machine q35, -accel hvf -accel tcg,tb-size=1024 -global PIIX4_PM.disable_s3=1 -global ICH9-LPC.disable_s3=1 -boot menu=on -m 4096 -device AC97 -name Debian-Desktop -usb -device usb-tablet,bus=usb-bus.0 -device usb-mouse,bus=usb-bus.0 -device usb-kbd,bus=usb-bus.0 -device ich9-usb-ehci1,id=usb-controller-0 -device ich9-usb-uhci1,masterbus=usb-controller-0.0,firstport=0,multifunction=on -device ich9-usb-uhci2,masterbus=usb-controller-0.0,firstport=2,multifunction=on -device ich9-usb-uhci3,masterbus=usb-controller-0.0,firstport=4,multifunction=on -chardev spicevmc,name=usbredir,id=usbredirchardev0 -device usb-redir,chardev=usbredirchardev0,id=usbredirdev0,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev1 -device usb-redir,chardev=usbredirchardev1,id=usbredirdev1,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev2 -device usb-redir,chardev=usbredirchardev2,id=usbredirdev2,bus=usb-controller-0.0 -device ide-hd,bus=ide.0,drive=drive0,bootindex=0 -drive if=none,media=disk,id=drive0,file=/Users/borja/Library/Containers/com.utmapp.UTM/Data/Documents/Debian-Desktop.utm/Images/labit-disk001.qcow2,cache=writethrough -device rtl8139,mac=C2:88:25:D8:67:C3,netdev=net0 -netdev vmnet-macos,mode=shared,id=net0 -device virtio-serial -device virtserialport,chardev=vdagent,name=com.redhat.spice.0 -chardev spicevmc,id=vdagent,debug=0,name=vdagent -uuid 81755D42-6831-410E-9FA8-A89B64FC1756 -rtc base=localtime
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: Started vmnet interface with configuration:
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: MTU:              1500
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: Max packet size:  1514
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: MAC:              52:85:89:78:8b:4f
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: DHCP IPv4 start:  192.168.64.1
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: DHCP IPv4 end:    192.168.64.254
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: IPv4 subnet mask: 255.255.255.0
qemu-x86_64-softmmu: -netdev vmnet-macos,mode=shared,id=net0: info: UUID:             F02A3C20-FE86-420A-8C76-A98F3C60727B
qemu-x86_64-softmmu: warning: global PIIX4_PM.disable_s3=1 not used
qemu-x86_64-softmmu: warning: Spice: playback:0 (0x7fb2e388d200): setsockopt failed, Operation not supported on socket
qemu-x86_64-softmmu: warning: Spice: record:0 (0x7fb2e388d150): setsockopt failed, Operation not supported on socket
vmx_write_mem: mmu_gva_to_gpa ffff88817b097000 failed

**Reply by conath** [reply]
Thanks. @osy any of these messages look like a UTM or QEMU bug to you? Borjand’s VM crashes after these messages:

```log
qemu-x86_64-softmmu: warning: Spice: playback:0 (0x7fb2e388d200): setsockopt failed, Operation not supported on socket
qemu-x86_64-softmmu: warning: Spice: record:0 (0x7fb2e388d150): setsockopt failed, Operation not supported on socket
vmx_write_mem: mmu_gva_to_gpa ffff88817b097000 failed
```

**Reply by Ganderse** [reply]
Hey @conath , I am trying to install a vm from vulnhub on UTM but I am having problems with it. I have converted the ova file into qcow2 as you suggested, but it still doesn't seem to work.

here is what i get when i start the machine

<img width="776" alt="Screenshot 2022-01-26 at 15 36 18" src="https://user-images.githubusercontent.com/93227573/151182851-2949e48b-a297-4872-bbd8-467828d6d6c6.png">

**Reply by alexiacl** [reply]
I am having the same problem. Have you found out what was happening?

**Reply by conath** [reply]
The VMs might be intended for legacy BIOS mode, so you can try disabling "UEFI Boot" in the System settings and see if it works.

**Reply by mtoseef99** [reply]
> The VMs might be intended for legacy BIOS mode, so you can try disabling "UEFI Boot" in the System settings and see if it works.

I am facing the same error, and after disabling the "UEFI Boot" under QEMU tab, I am getting the message below.

<img width="752" alt="image" src="https://user-images.githubusercontent.com/16485226/189134182-fc364069-c27c-40af-9e31-5850dc33459f.png">

Any suggestions?

**Reply by YYACMU** [reply]
> > The VMs might be intended for legacy BIOS mode, so you can try disabling "UEFI Boot" in the System settings and see if it works.
> 
> I am facing the same error, and after disabling the "UEFI Boot" under QEMU tab, I am getting the message below.
> 
> <img alt="image" width="752" src="https://user-images.githubusercontent.com/16485226/189134182-fc364069-c27c-40af-9e31-5850dc33459f.png">
> 
> Any suggestions?

You should try this.
- Change your disk mode to NVMe.
- Close the UEFI option in QEMU
- Start the machine and then press esc immediately
- Choose NVMe to start
- DONE!

**Reply by YYACMU** [reply]
> Hey @conath , I am trying to install a vm from vulnhub on UTM but I am having problems with it. I have converted the ova file into qcow2 as you suggested, but it still doesn't seem to work.
> 
> here is what i get when i start the machine
> 
> <img alt="Screenshot 2022-01-26 at 15 36 18" width="776" src="https://user-images.githubusercontent.com/93227573/151182851-2949e48b-a297-4872-bbd8-467828d6d6c6.png">

You should try this.
- Change your disk mode to NVMe.
- Close the UEFI option in QEMU
- Start the machine and then press esc immediately
- Choose NVMe to start
- DONE!

**Reply by J-GitHub1** [reply]
LEGEND

**Reply by BaTpy6KA** [reply]
turn of UEFI

**Reply by BaTpy6KA** [reply]
and move qcow2 ide disk up under other

**Comment by Foiler-NZ** [comment]
I converted a Virtualbox Windows XP instance from OVA to QCOW2 per the link above. 

Windows XP boots until just before the Windows login screen, then the VM crashes and tries to boot again .. the same thing happens in Safe Mode.  I presume I have to change the System settings or maybe the Display settings to get the boot process to complete?

**Reply by osy** [reply]
Honestly, you should just make a new install. Windows XP is notorious for not working with hardware changes. Back when we had physical computers running Windows XP, it's typical for it to stop booting when you change the RAM or HDD or something.

**Reply by conath** [reply]
Adding to what osy said. You can import the existing VM’s hard disk into the new VM after it’s installed and then access the data (shows up like any hard disk in Windows XP).

**Reply by Foiler-NZ** [reply]
I am quite keen to get that VM going as is due to previously licensed software .. in Safe Mode you can see that the boot process crashes after loading mup.sys .. which is sometimes caused by disk errors apparently .. I loaded the drive in another XP vm and ran a chkdsk but no errors showed up so it is something else .. any suggestions would be welcome :-)

**Reply by conath** [reply]
My suggestion would be to look into older forums for a similar situation - UTM is a relatively young piece of software, QEMU (its main backend) has been around for much longer. If you search online for "move VirtualBox to QEMU" or similar, you might find a solution that can be adapted to work with UTM. Good luck! 👍

**Reply by startergo** [reply]
So at this page it says it is enough to set ```uuid.bios``` to ```-uuid```  and  ```ethernet0.generatedAddress``` to ```macaddr```. Can someone confirm?
https://www.linux-kvm.org/page/How_To_Migrate_From_Vmware_To_KVM

> Look at your VMX configuration file:

scsi0:0.fileName = "zimbra-000001.vmdk"
uuid.bios = "56 4d 3f 3d 32 80 5b f2-94 31 21 c9 b2 c3 93 b9"
ethernet0.generatedAddress = "00:0c:29:c3:93:b9"
And then build the command-line:

kvm -drive file=zimbra-000001.vmdk,boot=on \
  -net nic,macaddr=00:0c:29:c3:93:b9 -net tap \
  -uuid 564d3f3d-3280-5bf2-9431-21c9b2c393b9

**Reply by YYACMU** [reply]
You should try this.
- Change your disk mode to NVMe.
- Close the UEFI option in QEMU
- Start the machine and then press esc immediately
- Choose NVMe to start
- DONE!

**Reply by belyaevEDU** [reply]
Hiya! Were you ever able to fix this?

**Comment by ghost** [comment]
This process solve my issue with UTM, hope it helps

1. Follow the tutorial shared by @conath to [convert .ova to .qcow2][1] (mac: brew install qemu - not qemu-utils)
2. Create New Virtual Machine
3. Emulate
4. Custom
5. Skip ISO boot
6. The rest of configs as your needs or default (continue, continue, etc... save)
7. Continue & Save

At this point something that looks like a virtual machine was created in the left panel.

1. Open the VM that you just created
2. Go to Setings
3. Drives: In the popup Settings window at the bottom left is Drives - Delete that Drive (Mine is called IDE Drive)
4. Drives: Click in New...
5. Import
6. Select the qcow2
7. Go to QEMU Tab
8. Unselect "UEFI Boot"
9. Save
10. Run 

This part was taken from [this video][2]

[1]: https://www.xmodulo.com/convert-ova-to-qcow2-linux.html
[2]: https://www.youtube.com/watch?v=enF3zbyiNZA

**Reply by kevinadhiguna** [reply]
I followed this steps and failed to boot Kali Linux:
`[ ... ] Failed to access perfctr msr (MSR c0010004 is 0)`

![image](https://github.com/user-attachments/assets/957cd635-2ff1-4bd6-a166-2ed478a0a425)

Any ideas?

**Reply by fs-c** [reply]
for me (booting a x64 linux image) disabling uefi boot made it hang during boot, but enabling it worked

**Comment by sameoj** [comment]
Hi,

I`m trying to login to my Window Server installed on UTM. 

Each time I press these keys: Command+Option+Esc; it displays "Force Quit Application." Please see attached screenshot.

Can anyone, please, tell me what I`m doing wrong?

Thank you in advance

![image](https://user-images.githubusercontent.com/127625165/224833119-bcd23e8e-4be2-42ea-abce-83939835225d.png)

**Comment by LucasKuhn** [comment]
https://github.com/utmapp/UTM/discussions/2521#discussioncomment-4110452
I followed @laugsg answer and it worked, but the performance was super slow 🐢 
Asking UTM to force multicore gave a **significant** performance boost: 
On the initial UTM screen, select the vm, click on top right to edit, then:
Sytem -> Force multicore 
<img width="979" alt="Screenshot 2023-05-18 at 8 10 55 AM" src="https://github.com/utmapp/UTM/assets/26127185/95c9d1cc-a635-407a-9245-8303c84e4e0a">

**Comment by AartSpilt** [comment]
Thanks for the info. I did manage to make and run an virtual machine. And also managed to install Kali. What I want to do is to run both the virtual machine from vulnhub (for example earth) and Kali. And with netdiscover I want to assess the ip adress user by the virtual machine. Anyone knows which network settings to use?
